### PR TITLE
Use  sasl_auth:client_done/1

### PR DESCRIPTION
**Pull request by starbelly**
sasl_auth was updated to require an explicit destruction of a context in an effort to avoid in memory race conditions. This commit merely updates the two plugins to use this function.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,12 +14,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        otp: ['23', '24', '25']
-        rebar3: ['3.16.1']
+        otp: ['25', '26', '27']
```
Comment (starbelly) [inline .github/workflows/ci.yml]: This is a norm for me, but can understand in the context of EMQ if you want to test further back. Let me know.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,12 +14,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        otp: ['23', '24', '25']
-        rebar3: ['3.16.1']
+        otp: ['24', '25', '26', '27']
```
Comment (starbelly) [inline .github/workflows/ci.yml:17]: Re-adding comment, we can still test 23, but IMHO we should drop it, and 24, though I left that.

File: .github/workflows/erlang.yml [hunk 3]
```diff
@@ -24,6 +24,10 @@ jobs:
       with:
         path: /tmp/.buildx-cache
         key: ${{ runner.os }}-buildx-${{ hashFiles('./example/image/Dockerfile') }}
+    - name: Install docker-compose
+      run: |
+        curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
```
Comment (starbelly) [inline .github/workflows/erlang.yml:29]: docker-compose doesn't seemed to be install anymore (although a plugin is), looking it up on docker github, this was the solution.

File: example/image/Dockerfile [hunk 4]
```diff
@@ -43,6 +43,8 @@ RUN curl -O https://raw.githubusercontent.com/kerl/kerl/master/kerl
 
 RUN chmod a+x kerl
 
+ENV KERL_BUILD_DOCS=
```
Comment (starbelly) [inline example/image/Dockerfile:46]: Makes build faster for end to end testing

File: rebar.config [hunk 5]
```diff
@@ -7,15 +7,16 @@
 ]}.
 
 {project_plugins, [
-    {rebar3_hex, "~> 7.0.1"},
-    {rebar3_format, "~> 1.0.1"},
-    {erlfmt, "~> 1.0.0"},
+    {pc, "~> 1.15.0"},
```
Comment (starbelly) [inline rebar.config:10]: Updated all per testing OTP 27 now.

File: src/brod_gssapi_v1.erl [hunk 6]
```diff
@@ -112,25 +112,28 @@ auth_begin(#{sasl_conn := Conn} = State) ->
...
     case send_sasl_token(State, Challenge) of
         {ok, _} ->
+            sasl_auth:client_done(Conn),
             set_sock_opts(State, [{active, once}]);
         Error ->
             Error
     end;
 auth_continue(#{handshake_vsn := 1, sasl_conn := Conn} = State, {sasl_continue, Challenge}) ->
-    case send_sasl_token(State, Challenge) of
+	case send_sasl_token(State, Challenge) of
```
Comment (zmstone) [inline src/brod_gssapi_v1.erl]:
```suggestion
    case send_sasl_token(State, Challenge) of
```

File: src/brod_gssapi_v1.erl [hunk 7]
```diff
@@ -112,25 +112,28 @@ auth_begin(#{sasl_conn := Conn} = State) ->
...
         {ok, Token} ->
             case sasl_auth:client_step(Conn, Token) of
                 {ok, SaslRes} ->
                     auth_continue(State, SaslRes);
                 Other ->
+                     sasl_auth:client_done(Conn),
```
Comment (zmstone) [inline src/brod_gssapi_v1.erl]:
```suggestion
                    sasl_auth:client_done(Conn),
```

File: rebar.config [hunk 8]
```diff
@@ -7,15 +7,16 @@
...
+    {erlfmt, "~> 1.3.0"},
     {rebar3_ex_doc, "~> 0.2.11"},
-    {rebar3_lint, "~> 1.0.2"},
-    {rebar3_hank, "~> 1.3.0"}
+    {rebar3_lint, "~> 3.2.6"},
+    {rebar3_hank, "~> 1.4.1"}
 ]}.
 
-{deps, [{sasl_auth, "2.1.1"}]}.
+{deps, [{sasl_auth, "2.3.0"}]}.
```
Comment (zmstone) [inline rebar.config]:
```suggestion
{deps, [{sasl_auth, "2.3.1"}]}.
```

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
published 0.1.3

**Pull request merged**